iam_policies.tf

```diff
@@ -189,26 +188,33 @@ resource "google_iap_web_backend_service_iam_policy" "jobs" {
   policy_data = data.google_iam_policy.iap_users.policy_data
 }
 
-resource "google_cloud_run_service_iam_binding" "frontend_run-invoker" {
+data "google_iam_policy" "run_users" {
+  binding {
+    role = "roles/run.invoker"
+    members = [
+        "serviceAccount:${google_project_service_identity.iap_sa.email}",
+        "serviceAccount:${google_service_account.pubsub_sa.email}",
+    ]
+  }
+}
+
+resource "google_cloud_run_service_iam_policy" "frontend_run-invoker" {
   location = google_cloud_run_service.frontend_run.location
   project = google_cloud_run_service.frontend_run.project
   service = google_cloud_run_service.frontend_run.name
-  role = "roles/run.invoker"
-  members = ["allUsers"]
+  policy_data = data.google_iam_policy.run_users.policy_data
 }
 
-resource "google_cloud_run_service_iam_binding" "controller_run-invoker" {
+resource "google_cloud_run_service_iam_policy" "controller_run-invoker" {
```

Comment:
We prefer using `google_cloud_run_service_iam_member` over policy as policy is authoritative.

Reply:
Noted. I've updated the PR with this.